src/Hint/List.hs

```diff
@@ -168,12 +170,12 @@ isAppend :: View a App2 => a -> Bool
 isAppend (view -> App2 op _ _) = varToStr op == "++"
 isAppend _ = False
 
-checks ::[(String, Bool -> LHsExpr GhcPs -> Maybe (LHsExpr GhcPs, [(String, R.SrcSpan)], String))]
-checks = let (*) = (,) in drop1 -- see #174
+checks :: [String] -> [(String, Bool -> LHsExpr GhcPs -> Maybe (LHsExpr GhcPs, [(String, R.SrcSpan)], String))]
+checks exts = let (*) = (,) in drop1 -- see #174
   [ "Use string literal" * useString
-  , "Use list literal" * useList
   , "Use :" * useCons
   ]
+  <> ["Use list literal" * useList | "OverloadedLists" `notElem` exts] -- see #114
```

Comment:
Can we pull out the existence or otherwise of OverloadedLists as a boolean and sent it down from the top, instead of just the full list of extensions.

Reply:
Thank you for the review. I've pushed 2 new commits to address both.